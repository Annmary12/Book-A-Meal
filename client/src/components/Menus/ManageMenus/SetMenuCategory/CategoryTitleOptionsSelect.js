import React from 'react';
import PropTypes from 'prop-types';
import categoryTitles from '../categoryTitles.js';
import styles from '../set-menu.scss';
import RemoveCategoryButton from './RemoveCategoryButton.js';

const CategoryTitleOptionsSelect = ({
  selectedTitles,
  handleChange,
  removeCategory,
  categoryCount,
  value,
  errors,
  categoryIndex
}) => {
  const error = errors[`categories[${categoryIndex}].title`];

  return (
    <label htmlFor="title" className={styles.label}>
      <div className={styles.title}>
      Category title:
        {
        categoryCount > 1 &&
        <RemoveCategoryButton
          handleRemoveCategory={removeCategory}
        />
      }
      </div>
      <select
        className={error ? styles.inputError : styles.input}
        name="title"
        onChange={handleChange}
        value={value}
      >
        <option value="" disabled>Select a category</option>
        {
        categoryTitles.map((title, index) => (
          <option
            key={index.toString()}
            value={title}
            disabled={selectedTitles.includes(title) ? 'disabled' : ''}
          >
            {title}
          </option>
        ))
      }
      </select>
      {
        error &&
        <div className={styles.errorMessage}>{error}</div>
      }
    </label>
  );
};

CategoryTitleOptionsSelect.propTypes = {
  selectedTitles: PropTypes.arrayOf(PropTypes.string).isRequired,
  handleChange: PropTypes.func.isRequired,
  value: PropTypes.string.isRequired,
  removeCategory: PropTypes.func.isRequired,
  categoryCount: PropTypes.number.isRequired,
  errors: PropTypes.objectOf(PropTypes.string).isRequired,
  categoryIndex: PropTypes.number.isRequired
};
export default CategoryTitleOptionsSelect;
