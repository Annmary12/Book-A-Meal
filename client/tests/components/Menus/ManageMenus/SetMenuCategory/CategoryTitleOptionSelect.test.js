import React from 'react';
import CategoryTitleOptionsSelect from '../../../../../src/components/Menus/ManageMenus/SetMenuCategory/CategoryTitleOptionsSelect';

const props = {
  selectedTitles: [],
  handleChange: jest.fn(),
  value: '',
  removeCategory: jest.fn(),
  categoryCount: 1,
  errors: {'categories[0].title': 'error'},
  categoryIndex: 0
}
describe('CategoryTitleOptionSelect component', () => {
  it('should match snapshot if category count is 1 and errors exist', () => {
    const wrapper = shallow(
      <CategoryTitleOptionsSelect {...props} />
    );

    expect(wrapper).toMatchSnapshot();
  });
  it('should match snapshot if category count greater than 1 and errors dont exist', () => {
    const wrapper = shallow(
      <CategoryTitleOptionsSelect
        {...props}
        errors={{}}
        categoryCount={2}
      />
    );

    expect(wrapper).toMatchSnapshot();
  })
});