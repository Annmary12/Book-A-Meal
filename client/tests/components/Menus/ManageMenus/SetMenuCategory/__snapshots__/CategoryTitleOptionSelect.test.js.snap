// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CategoryTitleOptionSelect component should match snapshot if category count greater than 1 and errors dont exist 1`] = `
<label
  className="label"
  htmlFor="title"
>
  <div
    className="title"
  >
    Category title:
    <RemoveCategoryButton
      handleRemoveCategory={[MockFunction]}
    />
  </div>
  <select
    className="input"
    name="title"
    onChange={[MockFunction]}
    value=""
  >
    <option
      disabled={true}
      value=""
    >
      Select a category
    </option>
    <option
      disabled=""
      key="0"
      value="Appetizer"
    >
      Appetizer
    </option>
    <option
      disabled=""
      key="1"
      value="Breakfast"
    >
      Breakfast
    </option>
    <option
      disabled=""
      key="2"
      value="Lunch"
    >
      Lunch
    </option>
    <option
      disabled=""
      key="3"
      value="Dinner"
    >
      Dinner
    </option>
    <option
      disabled=""
      key="4"
      value="Dessert"
    >
      Dessert
    </option>
    <option
      disabled=""
      key="5"
      value="Drinks"
    >
      Drinks
    </option>
    <option
      disabled=""
      key="6"
      value="Specials"
    >
      Specials
    </option>
  </select>
</label>
`;

exports[`CategoryTitleOptionSelect component should match snapshot if category count is 1 and errors exist 1`] = `
<label
  className="label"
  htmlFor="title"
>
  <div
    className="title"
  >
    Category title:
  </div>
  <select
    className="inputError"
    name="title"
    onChange={[MockFunction]}
    value=""
  >
    <option
      disabled={true}
      value=""
    >
      Select a category
    </option>
    <option
      disabled=""
      key="0"
      value="Appetizer"
    >
      Appetizer
    </option>
    <option
      disabled=""
      key="1"
      value="Breakfast"
    >
      Breakfast
    </option>
    <option
      disabled=""
      key="2"
      value="Lunch"
    >
      Lunch
    </option>
    <option
      disabled=""
      key="3"
      value="Dinner"
    >
      Dinner
    </option>
    <option
      disabled=""
      key="4"
      value="Dessert"
    >
      Dessert
    </option>
    <option
      disabled=""
      key="5"
      value="Drinks"
    >
      Drinks
    </option>
    <option
      disabled=""
      key="6"
      value="Specials"
    >
      Specials
    </option>
  </select>
  <div
    className="errorMessage"
  >
    error
  </div>
</label>
`;
